concept/type/AttributeType.java

```diff
@@ -80,6 +81,8 @@
         DATETIME(Encoding.ValueType.DATETIME);
 
         private final Encoding.ValueType encoding;
+        private Set<ValueType> comparables;
+        private Set<ValueType> assignables;
```

Comment:
cache the comparables and assignables as they would be computed over and over again - this is good because inside an enum, these sets will be computed once and used everywhere